src/persistence/history.cpp

```diff
@@ -43,23 +43,33 @@ MessageState getMessageState(bool isPending, bool isBroken)
     return messageState;
 }
 
-QString generatePeerIdString(ToxPk const& pk)
+QString generateAuthorIdString()
```

Comment:
I don't have a really good answer here, but I will note that this is pretty confusing. It's not clear when you call this that you have to bind a blob. You have to know what SQL the function is going to write to know what arguments to add.

An expecially bad exmaple is generateNewFileTransferQueries, that's pretttty tough to follow which arguments are binding where.

That being said the alternative is writing this same statement every time you want to get an author :\

Reply:
Agreed :S

All of the generate.*String functions in History got a lot messier to use in this change. I don't think there's really an easy fix, and don't know that copy pasting the sub-query statement 15 times across the file is much better, but maybe it is.

Reply:
Should be addressed by https://github.com/qTox/qTox/pull/6583#discussion_r835998682